phases/19-capstone-projects/68-rag-eval-precision-recall/code/main.py

```
@@ -0,0 +1,427 @@
+def reciprocal_rank(retrieved: list[str], gold: set[str]) -> float:
+    for i, doc_id in enumerate(retrieved):
+        if doc_id in gold:
+            return 1.0 / (i + 1)
+    return 0.0
+
+
+def mean_reciprocal_rank(retrieved_per_query: list[list[str]],
+                         gold_per_query: list[set[str]]) -> float:
+    if not retrieved_per_query:
+        return 0.0
+    return sum(reciprocal_rank(r, g) for r, g in zip(retrieved_per_query, gold_per_query)) / len(retrieved_per_query)
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)